interp_extrap_lib.R

```diff
@@ -0,0 +1,87 @@
+# Function to interpolate for each group (ID) in the data
+#' @param interp_type: Type of interpolation, either approx for linear or spline for nonlinear
+#' @param variable: Either "income" or "networth"
+#' @param grouped_df: data frame with year, variable to interpolate, and ID
+
+grouped_interpolate <- function(grouped_df, variable, interp_type = approx) {
+    if (variable == 'income'){
+        interp <- interp_type(x = grouped_df$year, y = grouped_df$income, xout = grouped_df$year)$y
+    }
+    else if (variable == 'networth'){
+        interp <- interp_type(x = grouped_df$year, y = grouped_df$networth, xout = grouped_df$year)$y
+    }
+    return(data.frame(year = grouped_df$year, interp) |>
+               rename(!!variable := interp))
+}
+
+
+# Function to reverse a time series for backcasting
+reverse_ts <- function(y)
+{
+    ts(rev(y), start=tsp(y)[1L], frequency=frequency(y))
+}
+# Function to reverse a forecast
+reverse_forecast <- function(object)
```

Comment:
It would be helpful to provide some comments here, in particular to describe attributes of `object` used in this function and to provide some intuition for lines 29-30 that reverse `object[["mean"]]`.

Reply:
Yes, I'll add some more annotation to explain what's going on here!